lib/matplotlib/tests/test_dates.py

```diff
@@ -500,12 +500,88 @@ def _create_auto_date_locator(date1, date2):
                 ],
                )
 
+    d1 = datetime.datetime(1997, 1, 1)
     for t_delta, expected in results:
         d2 = d1 + t_delta
         locator = _create_auto_date_locator(d1, d2)
         assert list(map(str, mdates.num2date(locator()))) == expected
 
 
+@pytest.mark.pytz
+@pytest.mark.skipif(not __has_pytz(), reason="Requires pytz")
```

Comment:
Why are you using `pytz` for this? Tests with `pytz` are just to ensure `matplotlib` is *compatible* with `pytz`, but this is a general functionality test. It's better to use `dateutil.tz`, which is not conditional on `pytz` installation.

Reply:
The original bug report was pytz, and pytz doesn't work if the code is not as written in the PR.  i.e. I needed to test pytz to get the PR to work, so it makes sense to test it here...

Reply:
Is it not possible to reproduce the bug with `dateutil.tz`? Just because the reporter used `pytz` doesn't mean you can't create an MWE without that dependency.

Reply:
Just checked, the original bug is also present when you use `dateutil.tz`, with the original MWE now (more accurately in this case):

```python
import matplotlib.pyplot as plt
from dateutil import tz
from datetime import datetime, timedelta

x = [datetime(2010, 1, 1, tzinfo=tz.gettz('America/New_York')) + timedelta(i)
     for i in range(2000)]
plt.plot(x, [0] * len(x));
```